src/Plugin/Block/FooterBlock.php

```diff
@@ -81,9 +93,59 @@ public function build() {
 
     NestedArray::setValue($build, ['#corporate_footer', 'bottom_links'], $config->get('bottom_links'));
 
+    $general_links = $this->retriveCustomFooterLinks('footer_general_link', $cache);
+    $social_links = $this->retriveCustomFooterLinks('footer_social_link', $cache);
+    $site_info_config = $this->configFactory->get('system.site');
+    $cache->addCacheableDependency($site_info_config);
+    $site_identity = $site_info_config->get('name');
+
+    if (!empty($site_identity) && !empty($social_links) && !empty($general_links)) {
+      NestedArray::setValue($build, ['#custom_footer', 'site_identity'], $site_identity);
+      NestedArray::setValue($build, ['#custom_footer', 'social_links'], $social_links);
+      NestedArray::setValue($build, ['#custom_footer', 'other_links'], $general_links);
+    }
+
```

Comment:
We need to always show the three elements?
Can't we have only, for example, **site identity** and **social links** in custom footer?

Reply:
I've asked this in comment of ticket.
This is a more requirement question.
From the ECL point of view, custom footer always visible if it has not empty at least **site identity**.
But this is not working for us. Because for site identity we use Site name (which mostly always visible).